peercred.go

```diff
@@ -29,6 +30,12 @@ func (c *Creds) UserID() (uid string, ok bool) {
 	return c.uid, c.uid != ""
 }
 
+// GroupID returns the groupid that owns the other side of the connection, if
+// known. (ok is false if not known)
+func (c *Creds) GroupID() (gid string, ok bool) {
```

Comment:
Not sure about this API. Maybe we should follow macOS/BSD and provide an accessor for all groups given that `Xucred.Groups` may contain more than one group ID, e.g.:

```go
func (c *Creds) Groups() (groups []string, ok bool)
```

and on Linux we'd just return the single gid we get in `cred.Gid`?

Reply:
Is this any different than user.LookupId + https://golang.org/pkg/os/user/#User.GroupIds ?

Any API giving me only 1 group feels weird, unless there's some reason for it... is this the primary group or something? What are primary groups used for?

Reply:
> Is this any different than user.LookupId + https://golang.org/pkg/os/user/#User.GroupIds ?

I don't think it's different from what `user.LookupId` + https://golang.org/pkg/os/user/#User.GroupIds would return, but why not expose the info if we already get it "for free" without additional calls.

> Any API giving me only 1 group feels weird, unless there's some reason for it... is this the primary group or something? What are primary groups used for?

Yeah, this is the primary group and on Linux it's the only one we get. But I agree it feels weird. On the BSDs and macOS it returns all the groups. So how about changing this to return all groups on BSD/macOS and on Linux use `user.LookupId` + `os/user/#User.GroupIds` to retrieve the complete list groups?

Reply:
It seems, on macOS at least not all groups are necessarily returns since len of `Xucred.Groups` is only 16. Given this shortcoming I think it doesn't make sense to expose this information. Closing the PR.